llvm/test/CodeGen/BPF/BTF/data-carrying-enum.ll

```diff
@@ -0,0 +1,110 @@
+; RUN: llc -mtriple=bpfel -filetype=obj -o %t1 %s
+; RUN: llvm-objcopy --dump-section='.BTF'=%t2 %t1
+; RUN: %python %p/print_btf.py %t2 | FileCheck -check-prefixes=CHECK-BTF %s
+; RUN: llc -mtriple=bpfeb -filetype=obj -o %t1 %s
+; RUN: llvm-objcopy --dump-section='.BTF'=%t2 %t1
+; RUN: %python %p/print_btf.py %t2 | FileCheck -check-prefixes=CHECK-BTF %s
+;
+; Source:
+;   #![no_std]
+;   #![no_main]
+;   
+;   pub enum DataCarryingEnum {
+;       First { a: u32, b: i32 },
+;       Second(u32, i32),
+;       Third(u32),
+;   }
+;   
+;   #[unsafe(no_mangle)]
+;   pub static X: DataCarryingEnum = DataCarryingEnum::First { a: 54, b: -23 };
+;   #[unsafe(no_mangle)]
+;   pub static Y: DataCarryingEnum = DataCarryingEnum::Second(54, -23);
+;   #[unsafe(no_mangle)]
+;   pub static Z: DataCarryingEnum = DataCarryingEnum::Third(36);
+;   
+;   #[cfg(not(test))]
+;   #[panic_handler]
+;   fn panic(_info: &core::panic::PanicInfo) -> ! {
+;       loop {}
+;   }
+; Compilation flag:
+;   cargo +nightly rustc -Zbuild-std=core --target=bpfel-unknown-none -- --emit=llvm-bc
+;   llvm-extract --glob=X --glob=Y --glob=Z $(find target/ -name "*.bc" | head -n 1) -o data-carrying-enum.bc
```

Comment:
This behavior can be reproduced only with Rust. However, this trick cleans up the noisy IR produces by Rust from the panic handler and all unnecessary core types. I hope it's good enough for the test.